src/Service/ImagesService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\ImageRepository;
+use BBC\ProgrammesPagesService\Domain\Entity\Image;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\ImageMapper;
+
+class ImagesService extends AbstractService
+{
+    /** @var ImageMapper */
+    protected $mapper;
+
+    /** @var ImageRepository */
+    protected $repository;
+
+    public function __construct(
+        ImageRepository $repository,
+        ImageMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    public function findByPid(Pid $pid, $ttl = CacheInterface::NORMAL): ?Image
```

Comment:
Did you say there might be multiple iSite promos? If so would we want to do findByPids so that we can request multiple images all in one query?

Reply:
There can be more than one. But I haven't found an instance when we want to query for more than one. I think we should wait until we need that before we add it.